rust/ip-packet/src/lib.rs

```diff
@@ -103,10 +795,73 @@ impl<'a> MutableIpPacket<'a> {
         for_both!(self, |i| i.get_destination().into())
     }
 
+    pub fn set_source_protocol(&mut self, v: u16) {
+        if let Some(mut p) = self.as_tcp() {
+            p.set_source(v);
+        }
+
+        if let Some(mut p) = self.as_udp() {
+            p.set_source(v);
+        }
+
+        self.set_icmp_identifier(v);
+    }
```

Comment:
I've made some changes to this in https://github.com/firezone/firezone/pull/5354/commits/ac8e2279332058fffca639b1a7e3a4b11c43d796#diff-13c5f91711cce55d8ba404f12513a706d5ba0842c765b2e976e702ad2022f90c. Could you cherry-pick those?

Reply:
Actually, we don't need this functions here, right? Let's remove them from this PR perhaps and add them in the next one, then you don't need to cherry-pick?

Reply:
I use them for the tests, I will cherry pick

Reply:
[24112fd](https://github.com/firezone/firezone/pull/5364/commits/24112fdae83bc297f22ab8f3415cd9342b6a7c3c)